pkg/client/proxy_backup.go

```diff
@@ -94,7 +94,12 @@ func (c *ProxyClient) BackupRestore(serviceAddress, url, target, volumeName stri
 	}
 
 	defer func() {
-		err = errors.Wrapf(err, "%v failed to restore backup %v to volume %v", c.getProxyErrorPrefix(serviceAddress), url, volumeName)
+		if err != nil {
+			_, ok := err.(TaskError)
```

Comment:
Finally know why you are doing this. 

Because the caller of this method will check the concert type of the returned error, you need to return the concrete one instead of another wrapped error.